Make container-interop a forwards-compatible polyfill for PSR-11 (#97)
container-interop/container-interop: README.md

[![Latest Stable Version](https://poser.pugx.org/container-interop/container-interop/v/stable.png)](https://packagist.org/packages/container-interop/container-interop)
[![Total Downloads](https://poser.pugx.org/container-interop/container-interop/downloads.svg)](https://packagist.org/packages/container-interop/container-interop)

## Polyfill Status

> Starting with the 1.3.0 release, the various artifacts of this package are all now [class aliases](https://php.net/class_alias) of the official [PSR-11 interfaces](https://www.php-fig.org/psr/psr-11/).

Reviewer: ```suggestion
Starting with the 2.0.0 release, the various artifacts of this package are all now [class aliases](https://php.net/class_alias) of the official [PSR-11 interfaces](https://www.php-fig.org/psr/psr-11/).
```

This version of the package can be used in interim releases of packages that previously depended on container-interop to allow them to adopt PSR-11 in a backwards compatible way, since aliases and the classes they alias can be used interchangeably.

We recommend the following workflow for **library packages**:

- Create a new minor release with the following changes to your package:

>   - Require `psr/container` (`composer require psr/container`)

Reviewer 2: ```suggestion
  - Require `psr/container` (`composer require psr/container:^1.1`)
```

>   - Update your minimum supported container-interop/container-interop version to 1.3.0 (`composer require "psr/container:^1.3"`)

Reviewer 2: ```suggestion
  - Update your minimum supported container-interop/container-interop version to 1.3.0 (`composer require "container-interop/container-interop:^1.3"`)
```

>   - Update your minimum supported container-interop/container-interop version to 1.3.0 (`composer require "container-interop/container-interop:^1.3"`)

Reviewer 2: I think we have to change this as well, right?

What I do actually wondering about is, how we ensure compatibility for both v1 and v2 here?
Can this be achieved?
Sadly thats months ago that I spent time on that front and thus I can't remember. Maybe I find some time after PHP 8.1 and stuff to outline this.

  - Update any typehints in your library that reference container-interop interfaces to instead reference PSR-11 interfaces:
    - `Interop\Container\ContainerInterface` becomes `Psr\Container\ContainerInterface`

>     - `Interop\Container\\Exception\ContainerException` becomes `Psr\Container\ContainerExceptionInterface`

Reviewer: ```suggestion
    - `Interop\Container\Exception\ContainerException` becomes `Psr\Container\ContainerExceptionInterface`
```
